sunpy/instr/goes.py

```diff
@@ -583,16 +698,533 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
+    # values of flux ratio
+    spline = interpolate.splrep(modeltemp, model_loss_rate, s=0)
+    rad_loss = em.value * interpolate.splev(temp.value, spline, der=0)
+    rad_loss = u.Quantity(rad_loss, unit='erg/s')
+    rad_loss = rad_loss.to(u.J/u.s)
+
+    # If obstime keyword giving measurement times is set, calculate
+    # radiative losses intergrated over time.
+    if obstime is not None:
+        # First ensure obstime is of same length as temp and em and of
+        # correct type.
+        n = len(temp)
+        if len(obstime) != n:
+            raise IOError("obstime must have same number of elements as "
+                          "temp and em.")
+        if type(obstime) == pandas.tseries.index.DatetimeIndex:
+            obstime = obstime.to_pydatetime
+        if any(type(obst) == str for obst in obstime):
+            parse_time(obstime)
+        if not all(type(obst) == datetime.datetime for obst in obstime):
+            raise TypeError("obstime must be an array-like whose elements are"
+                            " convertible to datetime objects.")
+        # Check elements in obstime in chronological order
+        chrono_check = obstime-np.roll(obstime, 1)
+        chrono_check = chrono_check[1:]
+        if not all(chrono_check > datetime.timedelta(0)):
+            raise ValueError(
+                "Elements of obstime must be in chronological order.")
+        # Next, get measurement times in seconds from time of first
+        # measurement.
+        obstime_seconds = np.array([(ot-obstime[0]).total_seconds()
+                                    for ot in obstime], dtype="float64")
+        # Finally, integrate using trapezoid rule
+        rad_loss_int = trapz(rad_loss.value, obstime_seconds)
+        rad_loss_int = u.Quantity(rad_loss_int, unit=rad_loss.unit*u.s)
+        # Calculate cumulative radiated energy in each GOES channel as
+        # a function of time.
+        rad_loss_cumul = cumtrapz(rad_loss, obstime_seconds)
+        rad_loss_cumul = u.Quantity(rad_loss_cumul, unit=rad_loss.unit*u.s)
+        # Enter results into output dictionary.
+        rad_loss_out = {"rad_loss_rate":rad_loss,
+                        "rad_loss_cumul" : rad_loss_cumul,
+                        "rad_loss_int":rad_loss_int}
+    else:
+        rad_loss_out = {"rad_loss_rate":rad_loss}
+
+    return rad_loss_out
+
+def calculate_xray_luminosity(goeslc):
+    """
+    Calculates GOES solar X-ray luminosity.
+
+    This function calculates the solar X-ray luminosity in the GOES
+    wavelength ranges (1-8 angstroms and 0.5-4 angstroms) based on the
+    observed GOES fluxes.  The units of the results are W.  The calculation
+    is made by simply assuming that the radiation is emitted isotropically,
+    i.e. is distributed over a spherical surface area with a radius equal to
+    the Sun-Earth distance.  Once the luminosity in each GOES passband is
+    found, they are returned in a new LightCurve object also containing the
```

Comment:
(minor) LightCurve should be <backtick>~sunpy.lightcurve.LightCurve<backtick> as it will make a link but only print `LightCurve`